### PR TITLE
Dev increase code coverage

### PR DESCRIPTION
**Pull request by am-ghaseminia**
# Description and Impact

Writing some unit tests for SettingViewModel, DataViewModel and MediasViewModel

#
# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor
- [x] Testing

### CODE REVIEW

File: features/backup/src/test/java/ir/mehdiyari/krypt/backup/DataViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    private val dispatcher = UnconfinedTestDispatcher(scheduler)
+
+    private val dataViewModel = DataViewModel(
+        backupRepository,
+        filesRepository,
+        dispatcher,
+        filesUtilities,
+        mediaStoreManager
+    )
+
```
Comment (mehdiyari) [inline features/backup/src/test/java/ir/mehdiyari/krypt/backup/DataViewModelTest.kt:47]: Hey Akram, Thanks for this great PR, is good to have `tearDown` method with `@After` annotation, to clear up the mocks. Here and for the rest of the tests.

File: features/backup/src/test/java/ir/mehdiyari/krypt/backup/DataViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                coVerify(exactly = 1) {
+                    mediaStoreManager.scanAddedMedia(listOf(expectedFilePath))
+                }
+            }
+        }
+
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+open class MainDispatcherRule constructor(
```
Comment (mehdiyari) [inline features/backup/src/test/java/ir/mehdiyari/krypt/backup/DataViewModelTest.kt:107]: Good to extract this class into separate file.

File: features/media/list/src/test/java/ir/mehdiyari/krypt/list/MediaViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    private val falleryBuilderProvider = mockk<FalleryBuilderProvider>(relaxed = true)
+    private val scheduler = TestCoroutineScheduler()
+    private val dispatcher = UnconfinedTestDispatcher(scheduler)
+
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule(dispatcher)
+
+    init {
+        Dispatchers.setMain(dispatcher)
+    }
```
Comment (mehdiyari) [inline features/media/list/src/test/java/ir/mehdiyari/krypt/list/MediaViewModelTest.kt:53]: If we have rule, why we need this line?

File: features/media/list/src/test/java/ir/mehdiyari/krypt/list/MediaViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            assertEquals(listOf(
+                SelectedMediaItems("path1", false),
+                SelectedMediaItems("path2", false),
+                SelectedMediaItems("path3", false)), awaitItem())
+        }
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+open class MainDispatcherRule constructor(
```
Comment (mehdiyari) [inline features/media/list/src/test/java/ir/mehdiyari/krypt/list/MediaViewModelTest.kt:83]: We can't duplicate this file across all modules. Good to have a shared module with a test name, but it's out of scope maybe of this PR, you can add a todo on this class which says we want to refactor this to a new module ASAP.

**Comment by am-ghaseminia** [comment]
I've resolved your all comments, 
for the last comment I extract MainDispatcherRule and test dependencies to a new module called "testing", so we can use it for shared logic related to testing

File: shared/testing/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    defaultConfig {
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+}
+
+
+dependencies {
+    implementation("androidx.core:core-ktx:1.8.0")
+    implementation("androidx.appcompat:appcompat:1.6.1")
+    implementation("com.google.android.material:material:1.9.0")
```
Comment (mehdiyari) [inline shared/testing/build.gradle.kts:17]: Why do we need these dependencies? If we need them please use the version catalog like the rest of the gradle files.

File: shared/testing/proguard-rules.pro [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (mehdiyari) [inline shared/testing/proguard-rules.pro:1]: You can delete this file with other proguard files for this module

**Review by mehdiyari — APPROVED** [review]

**Comment by mehdiyari** [comment]
Thanks, @am-ghaseminia 
All good, You can merge the PR with the `Create a merge commit` strategy.

**Pull request merged**